core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -90,52 +79,26 @@ protected void doStop() {
 
     @Override
     protected void doClose() {
-        FutureUtils.cancel(scheduledRoutingTableFuture);
-        scheduledRoutingTableFuture = null;
         clusterService.remove(this);
     }
 
-    /** make sure that a reroute will be done by the next scheduled check */
-    public void scheduleReroute() {
-        routingTableDirty = true;
+    /**
+     * make sure that a reroute will be done by the next scheduled check
+     */
+    public void reroute(String reason) {
+        performReroute(reason);
```

Comment:
I wanted to separate the method we override in our tests from the public method, will make this one final